pyteal/compiler/compiler.py

```diff
@@ -307,12 +294,12 @@ def compileTeal(
     # control flow graph, the optimizer requires context across block boundaries. This
     # is necessary for the dependency checking of local slots. Global slots, slots
     # used by DynamicScratchVar, and reserved slots are not optimized.
-    if options.optimize.scratch_slots:
+    if options.optimize.optimize_scratch_slots(version):
         options.optimize._skip_slots = collect_unoptimized_slots(
             subroutine_start_blocks
         )
         for start in subroutine_start_blocks.values():
-            apply_global_optimizations(start, options.optimize)
+            apply_global_optimizations(start, options.optimize, version)
```

Comment:
We need to start passing `version` since `optimize.optimize_scratch_slots()` is now requiring version.